include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -207,7 +207,19 @@ bellman_ford_shortest_paths(const graph<V, E, T>& graph,
       }
     }
   }
-
+  // Negative cycle detection by doing an additional pass in the graph
+  for (const auto& [edge_id, edge] : graph.get_edges()) {
+    const auto [u, v]{edge_id};
+    WEIGHT_T weight = get_weight(edge);
+    if (shortest_paths[u].total_weight !=
+            std::numeric_limits<WEIGHT_T>::max() &&
+        shortest_paths[u].total_weight + weight <
+            shortest_paths[v].total_weight) {
+      std::ostringstream error_msg;
+      error_msg << "Negative cycle detected in the graph.";
+      throw std::invalid_argument{error_msg.str()};
```

Comment:
Can we not pass the character literal directly to the constructor of `std::invalid_argument`:
```suggestion
      throw std::invalid_argument{"Negative cycle detected in the graph."};
```